### DISCUSSION TITLE
How do I connect to the openai Applicable Form API in version 2.0 similar to the original litellm?

### DISCUSSION THREAD

**Discussion by flyfox666** [open-webui/open-webui#2748] [General]
Removed 🚫 Bundled LiteLLM Support Deprecated: Migrate your LiteLLM config.yaml to a self-hosted LiteLLM instance. LiteLLM can still be added via OpenAI Connections. added via OpenAI Connections. Download the LiteLLM config.yaml from admin settings > database > export LiteLLM config.yaml.

The latest version of openwebui has removed the litellm bundle, but I have a lot of interfaces that were called via litellm before, how can I use them in the new version? Now I found Groq can be used, deepseek can be used. But other similar chatglm doesn't work. 

Many Thanks

**Comment by justinh-rahb** [comment]
Here's a guide to help you move your `litellm` setup out of the `open-webui` container and into its own separate container. This will allow you to use `litellm` independently and connect it to `open-webui` as needed.

1. **Find your `config.yaml` file**

Your `config.yaml` file should be located in the `open-webui` container. To find it, use the following command:
```bash
docker cp open-webui:/app/backend/data/litellm/config.yaml .
```
This will copy the `config.yaml` file from the `open-webui` container to your current directory.

1. **Run a standalone `litellm` container**

To run `litellm` in its own container, use the following `docker run` command:
```bash
docker run -d \
  -p 4000:4000 \
  --name litellm \
  -v ./config.yaml:/app/config.yaml \
  -e LITELLM_MASTER_KEY=sk-12345 \
  --restart always \
  ghcr.io/berriai/litellm:main-latest \
  --config /app/config.yaml --port 4000
```
Replace `./config.yaml` with the path to the `config.yaml` file you copied from the `open-webui` container. Set the `LITELLM_MASTER_KEY` environment variable to your API key value.

1. **Connect `litellm` to `open-webui`**

Once you have the `litellm` container running, you can connect it to the `open-webui` container. In the `open-webui` settings, add an "OpenAI" base URL under "Connections" with the value `http://host.docker.internal:4000/v1` and set the API key to your `LITELLM_MASTER_KEY` value (e.g., `sk-12345`).

By following these steps, you'll have moved your `litellm` setup out of the `open-webui` container and into its own separate container. You can now use `litellm` independently and connect it to `open-webui` as needed.

**Reply by nautat** [reply]
FYI, I had to use main-stable instead, because main-latest threw some errors.
